src/sentry/tasks/base.py

```diff
@@ -58,6 +65,60 @@ def __call__(self, decorated_task: Any) -> Any:
         return limited_func
 
 
+def taskworker_override(
+    celery_task_attr: Callable[P, R],
+    taskworker_attr: Callable[P, R],
+    namespace: str,
+    task_name: str,
+) -> Callable[P, R]:
+    def override(*args: P.args, **kwargs: P.kwargs) -> R:
+        rollout = 0
+        option_flag = f"taskworker.{namespace}.rollout"
+        if options.isset(option_flag):
+            rollout_map = options.get(option_flag)
+            rollout = rollout_map.get(task_name, 0)
+
+        random.seed(datetime.now().timestamp())
+        if rollout > random.random():
+            metrics.incr(
+                "taskbroker.rollout.shimmed", tags={"namespace": namespace, "task_name": task_name}
+            )
+            return taskworker_attr(*args, **kwargs)
+
+        return celery_task_attr(*args, **kwargs)
+
+    functools.update_wrapper(override, celery_task_attr)
+    return override
+
+
+def override_task(
+    celery_task: Task,
+    taskworker_task: TaskWorkerTask,
+    taskworker_config: TaskworkerConfig,
+    task_name: str,
+) -> Task:
+    """
+    This function is used to override SentryTasks methods with TaskWorkerTask methods
+    depending on the rollout percentage set in sentry options.
+
+    This is used to migrate tasks from celery to taskworker in a controlled manner.
+    """
+    replacements = {"delay", "apply_async"}
+    for attr_name in replacements:
+        celery_task_attr = getattr(celery_task, attr_name)
+        taskworker_attr = getattr(taskworker_task, attr_name)
+        if callable(celery_task_attr):
```

Comment:
```suggestion
        if callable(celery_task_attr) and callable(taskworker_attr):
```

Might be worth checking both attributes.